src/main/java/com/prajina/academy/controller/CourseController.java

```diff
@@ -41,13 +41,9 @@ public ModelAndView getCourses(HttpServletRequest request, HttpServletResponse r
 	@PostMapping("/save")
 	public ModelAndView save(HttpServletRequest request, HttpServletResponse response,
 			@ModelAttribute("course") Course course) {
-		// TODO This should be save/ merge when db is set up
-		if (service.isCourseExist(course)) {
-			service.update(course);
-		} else {
-			service.create(course);
-		}
-
+		
+		service.save(course);
```

Comment:
add a meaningful return to the save object, and return 
Use the service class to perform the save and retrieve in a @Transaction on the method.